provider/aspsms/aspsms.go

```diff
@@ -0,0 +1,75 @@
+package aspsms
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/messagebird/sachet"
+)
+
+// Config is the configuration struct for AspSms provider
+type Config struct {
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+}
+
+// AspSms contains the necessary values for the AspSms provider
+type AspSms struct {
+	Config
+}
+
+var aspsmsHTTPClient = &http.Client{Timeout: time.Second * 20}
```

Comment:
What do you think about embedding the HTTP client within the `AspSms` struct and creating it within `NewAspSms`?